askama_parser/src/expr.rs

```diff
@@ -927,6 +927,27 @@ impl<'a> Suffix<'a> {
             }
         }
 
+        fn lifetime<'a>(i: &mut InputStream<'a>) -> ParseResult<'a, ()> {
+            // Before the 2021 edition, awe can have whitespace characters between "r#" and the
+            // identifier so we allow it here.
+            let start = *i;
+            '\''.parse_next(i)?;
+            if alt((
+                (keyword("r#"), opt(skip_ws0), identifier).map(|(_, _, _)| ()),
+                (identifier, not(peek('#'))).map(|(_, _)| ()),
+            ))
+            .parse_next(i)
+            .is_err()
```

Comment:
Please don't ignore errors from the parser. The error location can be off. Use `opt(..).parse_next(i)?.is_none()` instead.